src/mono/mono/mini/interp/interp.c

```diff
@@ -3847,6 +3847,7 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 			frame->state.ip = ip + 6;
 			ves_pinvoke_method (csignature, (MonoFuncV)code, context, frame, &retval, save_last_error, cache, sp);
 
+			EXCEPTION_CHECKPOINT_GC_UNSAFE;
```

Comment:
Aren't we back in GC safe here?

Reply:
Followed the same pattern we did for `MINT_CALLI_NAT_FAST`.

Reply:
It's not clear to me why we're in GC unsafe there, though I guess if it throws we must be? @BrzVlad do you know what's going on here?

Reply:
Here is the story behind it https://github.com/mono/mono/pull/17016. You could also read the issue it mentioned in this PR. In short, usually the interpreter operates in GC Unsafe state.